src/web/assets/garnish/src/MenuBtn.js

```diff
@@ -268,6 +306,7 @@ export default Base.extend(
     destroy: function () {
       this.menu.destroy();
       this.$btn.removeData('menubtn');
+      this.observer = null;
```

Comment:
Is this enough to actually kill the observer? Guessing you’ll need to call `this.observer.disconnect()` for it to actually stop observing changes.

Reply:
Have added the disconnect before setting it to `null`.